lib/rubocop/cop/style/unpack_first.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for accessing the first element of *unpack* which can
+      # be replaced with the shorter method *unpack1*.
+      #
+      # @example
+      #
+      #   # bad
+      #   'foo'.unpack('h*').first
+      #   'foo'.unpack('h*')[0]
+      #   'foo'.unpack('h*').slice(0)
+      #   'foo'.unpack('h*').at(0)
+      #   'foo'.unpack('h*').take(1)
+      #
+      #   # good
+      #   'foo'.unpack1('h*')
+      #
+      class UnpackFirst < Cop
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.4
+
+        MSG = 'Use `String#unpack1(%<format>s)` instead of '\
+          '`String#unpack(%<format>s)%<method>s`.'.freeze
+
+        def_node_matcher :unpack_and_first_element?, <<-PATTERN
+          {
+            (send $(send (...) :unpack $(...)) :first)
+            (send $(send (...) :unpack $(...)) {:[] :slice :at} (int 0))
+            (send $(send (...) :unpack $(...)) :take (int 1))
+          }
+        PATTERN
+
+        def on_send(node)
+          unpack_and_first_element?(node) do |unpack_call, unpack_arg|
+            range =
+              Parser::Source::Range.new(node.loc.expression.source_buffer,
+                                        unpack_call.loc.expression.end_pos,
+                                        node.loc.expression.end_pos)
+            message = format(MSG,
+                             format: unpack_arg.source,
+                             method: range.source)
+            add_offense(node, message: message)
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.remove(node.loc.dot) if node.method_name == :first
```

Comment:
There are no tests for the auto-correction of `take`, `slice`, and `at`. I think the auto-correction is going to produce incorrect code when one of those is used.